x/tss/types/codec.go

```diff
@@ -7,53 +7,21 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
 	reward "github.com/axelarnetwork/axelar-core/x/reward/exported"
-	"github.com/axelarnetwork/axelar-core/x/tss/tofnd"
 )
 
 // RegisterLegacyAminoCodec registers concrete types on codec
 func RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
 	cdc.RegisterConcrete(&HeartBeatRequest{}, "tss/HeartBeatRequest", nil)
-	cdc.RegisterConcrete(&StartKeygenRequest{}, "tss/StartKeygen", nil)
-	cdc.RegisterConcrete(&ProcessKeygenTrafficResponse{}, "tss/KeygenTraffic", nil)
-	cdc.RegisterConcrete(&ProcessSignTrafficRequest{}, "tss/SignTraffic", nil)
-	cdc.RegisterConcrete(&RotateKeyRequest{}, "tss/RotateKey", nil)
-	cdc.RegisterConcrete(&VoteSigRequest{}, "tss/VoteSig", nil)
-	cdc.RegisterConcrete(&VotePubKeyRequest{}, "tss/VotePubKey", nil)
-	cdc.RegisterConcrete(&RegisterExternalKeysRequest{}, "tss/RegisterExternalKey", nil)
-	cdc.RegisterConcrete(&SubmitMultisigPubKeysRequest{}, "tss/SubmitMultisigPubKeys", nil)
-	cdc.RegisterConcrete(&SubmitMultisigSignaturesRequest{}, "tss/SubmitMultisigSignatures", nil)
 }
 
 // RegisterInterfaces registers types and interfaces with the given registry
 func RegisterInterfaces(registry cdctypes.InterfaceRegistry) {
 	registry.RegisterImplementations((*sdk.Msg)(nil),
 		&HeartBeatRequest{},
-		&StartKeygenRequest{},
-		&ProcessKeygenTrafficRequest{},
-		&ProcessSignTrafficRequest{},
-		&RotateKeyRequest{},
-		&VoteSigRequest{},
-		&VotePubKeyRequest{},
-		&RegisterExternalKeysRequest{},
-		&SubmitMultisigPubKeysRequest{},
-		&SubmitMultisigSignaturesRequest{},
-	)
-	registry.RegisterImplementations((*codec.ProtoMarshaler)(nil),
-		&tofnd.MessageOut_SignResult{},
-		&tofnd.MessageOut_KeygenResult{},
-		&tofnd.MessageOut_CriminalList{},
-		&QueryRecoveryResponse{},
-		&KeygenVoteData{},
 	)
 
 	registry.RegisterImplementations((*reward.Refundable)(nil),
 		&HeartBeatRequest{},
-		&ProcessKeygenTrafficRequest{},
-		&VotePubKeyRequest{},
-		&ProcessSignTrafficRequest{},
-		&VoteSigRequest{},
-		&SubmitMultisigPubKeysRequest{},
-		&SubmitMultisigSignaturesRequest{},
```

Comment:
Also an issue here. Refundable Tx uses the `Any` type. So, we need to keep the implementations registered under to not break tx parsing. But need to ensure that we don't refund when someone bundles these types into the tx in the current version.

Reply:
I'll create an issue for this to be tackled later. It requires some serious solution and can't fit in this PR.

Reply:
Yep, need to solve this carefully